platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java

```diff
@@ -29,10 +29,20 @@
 import org.gradle.jvm.toolchain.JavaToolchainSpec;
 
 /**
- * Common configuration for Java based projects. This is added by the {@link org.gradle.api.plugins.JavaBasePlugin}.
+ * Common configuration for Java based projects. This is added by the {@link JavaBasePlugin}.
+ *
+ * This extension would be more appropriately named the {@code JvmPluginExtension} extension.  It
+ * is used to configure all {@link org.gradle.jvm.component.internal.JvmSoftwareComponentInternal JvmSoftwareComponentInternal} components.  At present there
+ * is only one such component - the {@code java} component added by the {@link org.gradle.api.plugins.JavaPlugin JavaPlugin} - but
+ * multiple components may be created by JVM language plugins in the future.
```

Comment:
I don't think we should be including details about internal types in our public javadoc.

From the point of view of the user, there is only a single component currently and I don't think we need to say all this stuff about configuring all components or what happens if there are more than one component. Currently having more than one of these components is impossible with the current API we expose, so we shouldn't document that case. I think this can only confuse the user at the current moment

I think it is sufficient to explain that the `java` extension configures project defaults.